### PR TITLE
feat(ticket#2): work on login page UI design and validation with zod

### PR DESCRIPTION
**Pull request by nadirhussain786**
### Implementation

This PR adds the login page of our application. The enhancements include integrating the Shadcn UI components and implementing form validation using Zod. These changes aim to enhance both the user experience and the robustness of form handling.

### Resources
login paga

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import supabase from "@/lib/supabase/client";
+import Login from "../components/Login";
 
 export default function Home() {
   return (
-    <main className="flex min-h-screen flex-col items-center justify-between p-24">
-       Shoes Rack APP
+    <main className="min-h-screen bg-gradient-to-br from-gray-100 to-gray-200">
+       <Login />
```
Comment (Monfernape) [inline src/app/page.tsx:6]: Let's not add this here. Create a separate route `/login`

File: src/components/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
+"use client";
```
Comment (Monfernape) [inline src/components/Login.tsx:1]: Let's use component directory scarcely. In each route, you can have a component directory where components belonging to that route can reside.

File: src/components/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Eye, EyeOff, Lock, Mail } from "lucide-react";
+
+const formSchema = z.object({
+  email: z.string()
```
Comment (Monfernape) [inline src/components/Login.tsx:21]: We will not authenticating user against email. Try using phone number. Try using input masking of `0321-1234567`

File: src/components/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    .min(1, "Email is required")
+    .max(50, "Email must be at most 50 characters long")
+    .email("Invalid email address"),
+  password: z.string()
```
Comment (Monfernape) [inline src/components/Login.tsx:25]: Password validation is not strong enough.

File: src/components/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    .min(8, "Password must be at least 8 characters long")
+    .max(50, "Password must be at most 50 characters long")
+});
+
+type FormValues = z.infer<typeof formSchema>;
+
+export default function LoginPage() {
```
Comment (Monfernape) [inline src/components/Login.tsx:32]: Let's only use default for route component names.

File: src/lib/supabase/client.ts [hunk 6]
```diff
@@ -1,15 +0,0 @@
-import { createClient } from '@supabase/supabase-js';
-
-interface EnvVariables {
```
Comment (Monfernape) [inline src/lib/supabase/client.ts:3]: Let's not delete this. It'll be used.

**Review by Monfernape — CHANGES_REQUESTED** [review]
Let's take a look at comments and review this again.

File: src/app/page.tsx (hunk 1 above)
Comment (nadirhussain786) [reply src/app/page.tsx]: **Fixed**

File: src/components/Login.tsx (hunk 2 above)
Comment (nadirhussain786) [reply src/components/Login.tsx]: Fixed

File: src/components/Login.tsx (hunk 3 above)
Comment (nadirhussain786) [reply src/components/Login.tsx]: **Fixed**

File: src/components/Login.tsx (hunk 4 above)
Comment (nadirhussain786) [reply src/components/Login.tsx]: **Fixed** by using Regix

File: src/lib/supabase/client.ts (hunk 6 above)
Comment (nadirhussain786) [reply src/lib/supabase/client.ts:3]: Fixed

**Review by ghulamhafeez — APPROVED** [review]

**Review by Monfernape — APPROVED** [review]

**Pull request merged**